crates/subspace-core-primitives/src/lib.rs

```diff
@@ -77,6 +78,29 @@ const PUBLIC_KEY_LENGTH: usize = 32;
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 pub struct PublicKey([u8; PUBLIC_KEY_LENGTH]);
 
+// TODO: add std::error::Error implementation
+/// Unexpected length for public key
+#[derive(Debug, Clone, Copy)]
+pub struct UnexpectedLengthError;
+
+impl PublicKey {
+    /// Converts slice to public key
+    pub fn from_slice(bytes: &[u8]) -> Result<Self, UnexpectedLengthError> {
```

Comment:
Given that the bytes here are always from the pubkey in practice, perhaps it's simpler to use something like `unchecked_from` and expect inside the impl.

Reply:
Well, I mean in practice(the usages in this PR) `unchecked` already indicates you are aware of what kind of memory slice you are dealing with.

Reply:
Just a thought to simplify some matters, not important.